test/unit/math/src/mathComplexDouble.cpp

```diff
@@ -37,9 +37,18 @@ TEMPLATE_LIST_TEST_CASE("mathOpsComplexDouble", "[math] [operator]", TestAccFunc
     testTemplate.template operator()<alpaka::Complex<double>>();
 }
 
-#ifdef __cpp_lib_is_layout_compatible
-TEMPLATE_LIST_TEST_CASE("mathOpsComplexDouble", "[layout]", TestAccs)
+TEMPLATE_LIST_TEST_CASE("mathOpsComplexDouble", "[array-oriented]", TestAccs)
 {
-    STATIC_REQUIRE(std::is_layout_compatible_v<alpaka::Complex<double> std::complex<double>>);
+    // Ensure that our implementation matches the behaviour of std::complex with regard to array-oriented access.
+    // See https://en.cppreference.com/w/cpp/numeric/complex - Array-oriented access - for more information.
+    auto const complex_alpaka = alpaka::Complex<double>{42., 42};
+    auto const complex_std = std::complex<double>{42., 42.};
+
+    auto const real_alpaka = reinterpret_cast<double const(&)[2]>(complex_alpaka)[0];
+    auto const real_std = reinterpret_cast<double const(&)[2]>(complex_std)[0];
+    REQUIRE(alpaka::math::floatEqualExactNoWarning(real_alpaka, real_std));
+
+    auto const imag_alpaka = reinterpret_cast<double const(&)[2]>(complex_alpaka)[1];
+    auto const imag_std = reinterpret_cast<double const(&)[2]>(complex_std)[1];
+    REQUIRE(alpaka::math::floatEqualExactNoWarning(imag_alpaka, imag_std));
```

Comment:
```suggestion
    auto const complex_alpaka = alpaka::Complex<double>{42., 43};
    auto const complex_std = std::complex<double>{42., 43.};
    STATIC_REQUIRE(sizeof(complex_alpaka) == sizeof(double) * 2);
    STATIC_REQUIRE(sizeof(complex_alpaka) == sizeof(complex_std));
    REQUIRE(!std::memcmp(&complex_alpaka, &complex_std, sizeof(complex_std)));
```
I would rather just check their sizes and byte contents are the same. Also include the check that our complex is really just 2 doubles etc. So we know there is no padding anywhere.

Reply:
Good point about the padding. Right now we're not ensuring that our complex is not padded. This might be a problem with `long double` which our complex supports.

Reply:
I think you can keep your checks since they are also listed on cppreference, as discussed in #1720. But consider adding the check `STATIC_REQUIRE(sizeof(complex_alpaka) == sizeof(double) * 2);` so we know that we did not introduce any additional padding.

Reply:
Done.